### DISCUSSION TITLE
NodalGravity kernel

### DISCUSSION THREAD

**Discussion by chakra34** [idaholab/moose#24099] [Q&A Modules: General] [answered]
Hi Moose group,

I am currently comparing a linear elastic simulation with force boundary condition for a simple block geometry with dimensions (10mm x 10mm x 10mm) between ANSYS and Moose. So the geometry looks like:
![image](https://user-images.githubusercontent.com/22983992/232588797-fadbbe58-5019-42ed-b296-19127cd4a517.png)

And I am keeping the bottom surface fixed displacement and applying a force on the top selected nodes (node set) to basically mimic the following ansys problem:
![image](https://user-images.githubusercontent.com/22983992/232589027-e08eb5f0-5499-4ae7-adbc-b6e950973969.png)

Now the force in ansys is 10N (-y direction), and when I looked into the NodalGravity kernel which allows for a force boundary condition I saw that it sums up the force from all the nodes in the node set. Since my node set has 19 nodes, I impose the following (g = 10/19 = 0.5263 ;  and nodal mass = 1.0) in the input block:

```
[NodalKernels]
  [./force_y2]
    type = NodalGravity
    variable = disp_y
    boundary = 'top_load_curve'
    gravity_value = -0.5263 # inverse of nodal mass at cantilever end
    mass = 1.0 # commented out for testing purposes
  [../]
[]
```
But when I compare the results, I see a difference in the values between ansys and moose both in stress and displacement.

![image](https://user-images.githubusercontent.com/22983992/232589839-634e93da-3b03-4f75-821f-c3cb96865f0c.png)

Is there something I am missing, I can't seem to figure out what might be the issue.

Thanks again,

-Aritra

**Comment by hugary1995** [comment, marked as answer]
By comparing stresses, you are essentially chasing a singularity here. The output values of stresses are very sensitive to the method used to average/extrapolate cell values onto the nodes. I recommend you to compare reaction forces on the nodes rather than stress values. Alternatively, compute and compare the surface integral of the vonmises stress.

**Reply by chakra34** [reply]
Thanks a lot. Actually I also compared the displacements at the corresponding locations, and they are also different. On another note, I was wondering whether the way I am adding the nodal gravity as a force boundary condition makes sense ? Here is the displacement comparison:

![image](https://user-images.githubusercontent.com/22983992/232608183-15c782e9-a13b-4284-9361-e38c1653cf2b.png)

**Reply by hugary1995** [reply]
I guess ansys is handling this BC differently than your nodal kernel approach in moose. For example, ansys could be integrating the BC on a lower dimensional sideset. Another possibility is that ansys might be using a finer mesh. To examine the later, you can try refining your mesh.

**Reply by hugary1995** [reply]
There is also no guarantee that ansys was distributing the 10N force uniformly as what you did in moose. It might be simply adding a constraint that the total force on the sideset add up to 10N.
